### PR TITLE
fix: allow access to '..' in mapfs

### PR DESCRIPTION
**Pull request by nikpivkin**
## Description
The path “../” becomes “..” after cleaning, causing the MapFS [check](https://github.com/aquasecurity/trivy/blob/main/pkg/mapfs/fs.go#L148) that the path lies outside FS to fail.

### Before
```bash
trivy conf test/main.tf
2024-09-23T16:44:35+06:00       INFO    [misconfig] Misconfiguration scanning is enabled
2024-09-23T16:44:37+06:00       INFO    [terraform scanner] Scanning root module        file_path="."
2024-09-23T16:44:37+06:00       ERROR   [terraform evaluator] Failed to load module. Maybe try 'terraform init'?        err="file does not exist"
2024-09-23T16:44:37+06:00       INFO    Detected config files   num=1
```

### After
```bash
./trivy conf test/main.tf
2024-09-23T17:23:36+06:00       INFO    [misconfig] Misconfiguration scanning is enabled
2024-09-23T17:23:37+06:00       INFO    [terraform scanner] Scanning root module        file_path="."
2024-09-23T17:23:37+06:00       INFO    Detected config files   num=2

../main.tf (terraform)

Tests: 10 (SUCCESSES: 1, FAILURES: 9, EXCEPTIONS: 0)
Failures: 9 (UNKNOWN: 0, LOW: 2, MEDIUM: 1, HIGH: 6, CRITICAL: 0)

HIGH: No public access block so not blocking public acls
═════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════

S3 buckets should block public ACLs on buckets and any objects they contain. By blocking, PUTs with fail if the object has any public ACL a.


See https://avd.aquasec.com/misconfig/avd-aws-0086
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 ../main.tf:1-3
   via main.tf:1-4 (module.test)
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 ┌ resource "aws_s3_bucket" "name" {
   2 │   bucket = "root"
   3 └ }
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
....
```

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/7535

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/mapfs/fs.go [hunk 1]
```diff
@@ -245,3 +245,7 @@ func cleanPath(path string) string {
 	path = strings.TrimLeft(path, "/") // Remove the leading slash
 	return path
 }
+
+func (m *FS) isPathAboveRoot(name string) bool {
+	return strings.HasPrefix(name, "..") && m.underlyingRoot != ""
```
Comment (knqyf263) [inline pkg/mapfs/fs.go:250]:
```suggestion
	return name == ".." || strings.HasPrefix(name, "../") && m.underlyingRoot != ""
```

A regular file can have `..` prefix.

```
$ touch ..foo
$ ls -al
total 0
drwxr-xr-x  3 teppei   96  9 25 11:06 ./
drwxrwxrwt 37 root   1184  9 25 11:06 ../
-rw-r--r--  1 teppei    0  9 25 11:06 ..foo
```

Comment (nikpivkin) [reply pkg/mapfs/fs.go]: Fixed https://github.com/aquasecurity/trivy/pull/7575/commits/a0023203081dbf101ac4a0d48f7ee570f3018e2d

File: pkg/mapfs/fs_test.go [hunk 2]
```diff
@@ -478,3 +478,26 @@ func TestFS_RemoveAll(t *testing.T) {
 		require.ErrorIs(t, err, fs.ErrNotExist)
 	})
 }
+
+func TestFS_WithUnderlyingRoot(t *testing.T) {
+	root := "testdata/subdir"
```
Comment (knqyf263) [inline pkg/mapfs/fs_test.go:483]: Is there any reason why `testdata` is used? Can we use `filepath.Join(t.TempDir(), "subdir"`?

Comment (nikpivkin) [reply pkg/mapfs/fs_test.go:483]: The `testdata/subdir` contains some files to which access is tested.

Comment (knqyf263) [reply pkg/mapfs/fs_test.go:483]: OK, thanks.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**